src/Controller/Component/PublicationComponent.php

```diff
@@ -61,34 +63,39 @@ public function initialize(array $config): void
     {
         parent::initialize($config);
 
-        $this->loader = new TreeLoader($this->Objects->getLoader());
-
+        $objectsLoader = $this->Objects->getLoader();
+        $this->loader = new TreeLoader($objectsLoader);
         $this->loadModel('Trees');
 
         $publicationUname = $this->getConfig('publication');
         if (empty($publicationUname)) {
             throw new InvalidArgumentException('Missing configuration for root folder');
         }
 
-        $menuFolders = $this->getConfig('menuFolders', []);
-        $uname = array_merge([$publicationUname], array_values($menuFolders));
-        $folders = $this->Objects->loadObjects(compact('uname'), 'folders')
-            ->indexBy('uname')
-            ->toArray();
+        $publicationLoader = $objectsLoader;
+        if ($this->getConfig('publicationLoader') !== null) {
+            $publicationLoader = new ObjectsLoader(
+                $this->getConfig('publicationLoader.objectTypesConfig', []),
+                $this->getConfig('publicationLoader.autoHydrateAssociations', [])
+            );
+        }
 
-        if (!isset($folders[$publicationUname])) {
+        $publication = $publicationLoader->loadObject($publicationUname, 'folders');
+        if (empty($publication)) {
             throw new NotFoundException(__('Root folder does not exist: {0}', $publicationUname));
         }
```

Comment:
I believe `ObjectsLoader::loadObject()` throws a [`Cake\Datasource\Exception\RecordNotFoundException`](https://api.cakephp.org/3.5/class-Cake.Datasource.Exception.RecordNotFoundException.html) when the requested id or uname does not exist, rather than returning an empty value.

I agree that the exception message should be more specific in this case, though:

```suggestion
        try {
            $publication = $publicationLoader->loadObject($publicationUname, 'folders');
        } catch (RecordNotFoundException $e) {
            throw new NotFoundException(__('Root folder does not exist: {0}', $publicationUname), null, $e);
        }
```

Reply:
oh that's true, thanks for pointing out